rnd/autogpt_server/autogpt_server/data/block.py

```diff
@@ -0,0 +1,130 @@
+import inspect
+import json
+
+from abc import ABC, abstractmethod
+from prisma.models import AgentBlock
+from pydantic import BaseModel
+from typing import ClassVar
+
+
+class Block(ABC, BaseModel):
```

Comment:
This is the interface for the agent block (or `component` whatever you call it :))

it will require the input & output schema, and run function (that can both be async or sync).

We might also need to add more signature for schema docs/description (we can do it later).